test/async-emitter.js

```diff
@@ -0,0 +1,252 @@
+
+  ae.emit('e1');
+
+  test.strictSame(ae.wrappers.size, 0);
+  test.strictSame(ae.events.size, 0);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter count', async test => {
+  const ae = new AsyncEmitter();
+
+  ae.on('e1', async () => {});
+  ae.on('e1', async () => {});
+
+  test.strictSame(ae.count('e1'), 2);
+  test.strictSame(ae.count('e2'), 0);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter clear all', async test => {
+  const ae = new AsyncEmitter();
+
+  const fn = test.mustNotCall();
+
+  ae.on('e1', fn);
+  ae.clear();
+  ae.emit('e1');
+
+  test.strictSame(ae.wrappers.size, 0);
+  test.strictSame(ae.events.size, 0);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter clear by name', async test => {
+  const ae = new AsyncEmitter();
+
+  const fn = test.mustNotCall();
+
+  ae.on('e1', fn);
+  ae.clear('e1');
+  ae.clear('e2');
+  ae.emit('e1');
+
+  test.strictSame(ae.wrappers.size, 0);
+  test.strictSame(ae.events.size, 0);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter clear once', async test => {
+  const ae = new AsyncEmitter();
+
+  const f1 = test.mustNotCall();
+  const f2 = test.mustNotCall();
+
+  ae.once('e1', f1);
+  ae.once('e2', f2);
+
+  test.strictSame(ae.wrappers.size, 2);
+  test.strictSame(ae.events.size, 2);
+
+  ae.clear('e1');
+
+  test.strictSame(ae.wrappers.size, 1);
+  test.strictSame(ae.events.size, 1);
+
+  ae.emit('e1');
+
+  test.strictSame(ae.wrappers.size, 1);
+  test.strictSame(ae.events.size, 1);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter names', async test => {
+  const ae = new AsyncEmitter();
+
+  ae.on('e1', () => {});
+  ae.on('e1', () => {});
+  ae.on('e2', () => {});
+  ae.on('e3', () => {});
+
+  test.strictSame(ae.names(), ['e1', 'e2', 'e3']);
+
+  test.strictSame(ae.wrappers.size, 0);
+  test.strictSame(ae.events.size, 3);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter listeners', async test => {
+  const ae = new AsyncEmitter();
+
+  ae.on('e1', () => {});
+  ae.on('e1', () => {});
+  ae.on('e2', () => {});
+  ae.on('e3', () => {});
+
+  test.strictSame(ae.listeners('e1').size, 2);
+  test.strictSame(ae.listeners('e2').size, 1);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter await', async test => {
+  const ae = new AsyncEmitter();
+
+  const fn = test.mustCall();
+
+  ae.on('e1', fn);
+  await ae.emit('e1');
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter await multiple listeners', async test => {
+  const ae = new AsyncEmitter();
+
+  const f1 = test.mustCall();
+  ae.on('e1', f1);
+
+  const f2 = test.mustCall();
+  ae.on('e1', f2);
+
+  const f3 = test.mustCall();
+  ae.on('e1', f3);
+
+  await ae.emit('e1');
+
+  test.strictSame(ae.wrappers.size, 0);
+  test.strictSame(ae.events.size, 1);
+
+  test.end();
+});
+
+metatests.test('AsyncEmitter await multiple events', async test => {
+  const ae = new AsyncEmitter();
+
+  const f1 = test.mustCall();
+  ae.on('e1', f1);
+
+  const f2 = test.mustCall();
+  ae.on('e2', f2);
```

Comment:
Why not inline `f1`,`f2`... here and above?

Reply:
Fixed where possible